src/ComplaintDetail.jsx

```diff
@@ -1,11 +1,242 @@
+import PropTypes from 'prop-types'
 import React from 'react';
+import { FormattedDate } from 'react-intl';
+import './ComplaintDetail.less';
+
+const ERROR = 'ERROR'
+const WAITING = 'WAITING'
+const RESULTS = 'RESULTS'
 
 export default class ComplaintDetail extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      phase: WAITING
+    }
+
+    // Render/Phase Map
+    this.renderMap = {
+      ERROR: this._renderError.bind(this),
+      WAITING: this._renderWaiting.bind(this),
+      RESULTS: this._renderResults.bind(this),
+    }
+
+    // Binds
+    this._processError = this._processError.bind(this)
+    this._processResults = this._processResults.bind(this)
+  }
+
+  // --------------------------------------------------------------------------
+  // Interfacing with the API
+
+  _callAPI(id) {
```

Comment:
Not going to necessarily say that this would hold up a merge here, but might be worth considering whether these results should be in the global store or not.